hive-hooks/src/main/java/com/airbnb/reair/hive/hooks/MetastoreAuditLogHook.java

```diff
@@ -0,0 +1,382 @@
+
+      Set<WriteEntity> writeEntities = new HashSet<>();
+
+      writeEntities.add(
+        new WriteEntity(
+          new Table(event.getNewTable()),
+          WriteType.INSERT
+        )
+      );
+
+      run(readEntities, writeEntities, HiveOperation.ALTERTABLE);
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Listener which fires when a partition is added.
+   *
+   * The event only provides the partition which was added and the associated
+   * table, hence for auditing purposes the read/write differential is the
+   * non-presence and presence of the partition respectively.
+   *
+   * @param event The add partition event
+   */
+  @Override
+  public void onAddPartition(AddPartitionEvent event) throws MetaException {
+    try {
+      Table table = new Table(event.getTable());
+      Set<ReadEntity> readEntities = new HashSet<>();
+      readEntities.add(new ReadEntity(table));
+
+      Set<WriteEntity> writeEntities = new HashSet<>();
+      writeEntities.add(new WriteEntity(table, WriteType.INSERT));
+
+      for (org.apache.hadoop.hive.metastore.api.Partition partition :
+        event.getPartitions()) {
+        writeEntities.add(
+          new WriteEntity(
+            new Partition(table, partition),
+            WriteType.INSERT
+          )
+        );
+      }
+
+      run(readEntities, writeEntities, HiveOperation.ALTERTABLE_ADDPARTS);
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Listener which fires when a partition is dropped.
+   *
+   * The event only provides the partition which was dropped and the associated
+   * table, hence for auditing purposes the read/write differential is the
+   * presence and non-presence of the partition respectively.
+   *
+   * @param event The drop partition event
+   */
+  @Override
+  public void onDropPartition(DropPartitionEvent event) throws MetaException {
+    try {
+      Table table = new Table(event.getTable());
+
+      Set<ReadEntity> readEntities = new HashSet<>();
+      readEntities.add(new ReadEntity(table));
+
+      readEntities.add(
+        new ReadEntity(
+          new Partition(table, event.getPartition())
+        )
+      );
+
+      Set<WriteEntity> writeEntities = new HashSet<>();
+      writeEntities.add(new WriteEntity(table, WriteType.INSERT));
+
+      run(readEntities, writeEntities, HiveOperation.ALTERTABLE_DROPPARTS);
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Listener which fires when a partition is altered.
+   *
+   * For auditing purposes the read/write differential is the old and new
+   * partition respectively.
+   *
+   * @param event The add partition event
+   */
+  @Override
+  public void onAlterPartition(AlterPartitionEvent event) throws MetaException {
+    try {
+
+      // Table is invariant and thus arbitrary choice between old and new.
+      Table table = new Table(
```

Comment:
Sadly no. I'm somewhat perplexed as to why. 

https://hive.apache.org/javadocs/r0.13.1/api/metastore/org/apache/hadoop/hive/metastore/events/AlterPartitionEvent.html